crates/bevy_ecs/src/event.rs

```diff
@@ -164,6 +164,8 @@ pub struct EventReader<'w, 's, T: Resource> {
     events: Res<'w, Events<T>>,
 }
 
+unsafe impl<A, T: Resource> ReadOnlySystemParamFetch for EventReaderState<A, T> {}
```

Comment:
I'd like to see this change duplicated in another PR so it doesn't get held up.

Reply:
I mean, this change is a horrific hack. I don't even like that I can refer to `EventReaderState` here to add an implementation to it.

Ideally, we'd have `#[system_param(read_only)]` for a checked version of this, but I haven't implemented that. Someone is more than welcome to add this.

Reply:
Yeah okay, that's very fair. Never mind!

Reply:
This is https://github.com/bevyengine/bevy/pull/4650